File: 프로그래머스/월간 코드 챌린지 시즌1/스타 수열/스타 수열.py
def solution(a):
    cnts = [0 for i in range(len(a))]
    for ai in a: cnts[ai] += 1

    answer = 0
    for i in range(len(a)):
        if cnts[i] <= answer: continue

        length,j = 0,0
        while j < len(a)-1:
            if (a[j]==i or a[j+1]==i) and a[j]!=a[j+1]:
                length,j = length+1,j+1
            j += 1

        answer = max(length, answer)

    return answer*2

File: 프로그래머스/월간 코드 챌린지 시즌1/스타 수열/스타 수열.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> a) {
    // 각 숫자 개수 세기
    vector<int> cnts(a.size(), 0);
    for(int i=0; i<a.size(); i++)
        cnts[a[i]]++;
    
    int answer = 0;
    for(int i=0; i<a.size(); i++){
        // i 개수가 최댓값보다 작으면 건너뛰기
        if(cnts[i] <= answer) continue;
        
        // 스타 수열이 최대 길이 구하기
        int len = 0;
        for(int j=0; j<a.size()-1; j++){
            if((a[j]==i || a[j+1]==i) && a[j]!=a[j+1]){
                len++; j++;
            }
        }
        
        answer = max(len, answer);
    }
    
    return answer*2;
}
